routes/posts.js

```diff
@@ -0,0 +1,91 @@
+const express = require("express");
+const router = express.Router();
+const passport = require("passport");
+
+const Post = require("../models/Post");
+const User = require("../models/User");
+
+router.get("/", (req, res) => {
+  Post.find()
+    // get by latest
+    .sort({date: -1 })
+    .then(posts => {
+      res.json(posts);
+    })
+    .catch(err => {
+      res.status(500).json({error: err.message});
+    })
+})
+
+router.post("/", passport.authenticate("jwt", {session: false}), (req, res) => {
+  const { title, content} = req.body;
+  const currentUserId = req.user.id;
+  Post.create({title, content, user: currentUserId})
+    .then(post => {
+      res.json(post);
+    })
+    .catch(err => {
+      res.status(500).json({error: err.message});
+    })
+})
+
+router.get("/:id", (req, res) => {
+  const { id } = req.params;
+  Post.findById(id)
+    .then(post => {
+      res.json(post);
+    })
+    .catch(err => {
+      res.status(500).json({msg: "Can't find by that ID"});
+    })
+})
+
+router.delete("/:id", passport.authenticate("jwt", {session: false}), (req, res) => {
+  const { id } = req.params;
+  const currentUserId = req.user.id;
+  console.log("currentUser : ", req.user);
+  console.log("id params", id);
+  Post.findById(id)
+    .then(post => {
+      console.log("post that you get back", post);
+      if(post.user.toString() !== currentUserId) {
```

Comment:
Clever! Took me a bit to figure out what `toString()` is doing here and why you weren't calling `populate`, but I think this is much more direct (and maybe more secure?) Nice work 👍